core/string/ustring.cpp

```diff
@@ -3538,15 +3540,18 @@ String String::format(const Variant &values, const String &placeholder) const {
 				} else {
 					ERR_PRINT(String("STRING.format Inner Array size != 2 ").ascii().get_data());
 				}
-			} else { //Array structure ["RobotGuy","Logis","rookie"]
-				Variant v_val = values_arr[i];
-				String val = v_val;
+				continue;
+			}
+#endif // DISABLE_DEPRECATED
 
-				if (placeholder.find("_") > -1) {
-					new_string = new_string.replace(placeholder.replace("_", i_as_str), val);
-				} else {
-					new_string = new_string.replace_first(placeholder, val);
-				}
+			// Array structure ["RobotGuy", "Logis", "rookie"].
+			Variant v_val = values_arr[i];
+			String val = v_val;
+
+			if (placeholder.find("_") > -1) {
+				new_string = new_string.replace(placeholder.replace("_", i_as_str), val);
+			} else {
+				new_string = new_string.replace_first(placeholder, val);
```

Comment:
Should we move this code to be the first condition.
if (values_arr[i].get_type() == Variant::STRING) {
...
}
#ifndef DISABLE_DEPRECATED
			if (values_arr[i].get_type() == Variant::ARRAY) {
}